Dockerfile.prod

```diff
@@ -0,0 +1,42 @@
+# ----------------------------------------------------------------
+# This Dockerfile uses multiple best practices recommended by Snyk,
+# OWASP, and Docker. In this file the practices are defined, not 
+# explained. For more in depth explaination refer to the
+# security-documentation repository.
+#
+# This file should be used for production purposes. If building
+# an image for development use Dockerfile.dev
+# ----------------------------------------------------------------
+# ----------------------------------------------------------------
+# Build Stage
+# ----------------------------------------------------------------
+FROM node:latest as build
+WORKDIR /usr/src/app
+COPY package.json tsconfig.json /usr/src/app/
+RUN npm install
+
+COPY  ./src /usr/src/app/src
+COPY  ./docs /usr/src/app/docs
```

Comment:
Are the docs compiled? I am not sure what happens to swaggerdocs on compilation. If they aren't needed then after the compilation then we can just copy over the compiled /dist and the /nodemodules to the "serve" stage